packages/marshal/src/helpers/safe-promise.js

```diff
@@ -23,25 +23,52 @@ const checkPromiseOwnKeys = (pr, check) => {
     key => typeof key !== 'symbol' || !hasOwnPropertyOf(Promise.prototype, key),
   );
 
+  const checkSafeEnoughKey = key => {
+    const val = pr[key];
+    if (val === undefined || typeof val === 'number') {
+      return true;
+    }
+    if (
+      typeof val === 'object' &&
+      val !== null &&
+      isFrozen(val) &&
+      getPrototypeOf(val) === Object.prototype
+    ) {
+      const subKeys = ownKeys(val);
+      if (subKeys.length === 0) {
+        return true;
+      }
+
+      // At the time of this writing, Node's async_hooks contains the
+      // following code, which we can also safely tolerate
+      //
+      // function destroyTracking(promise, parent) {
+      // trackPromise(promise, parent);
+      //   const asyncId = promise[async_id_symbol];
+      //   const destroyed = { destroyed: false };
+      //   promise[destroyedSymbol] = destroyed;
+      //   registerDestroyHook(promise, asyncId, destroyed);
+      // }
+
+      if (
+        subKeys.length === 1 &&
+        subKeys[0] === 'destroyed' &&
+        val.destroyed === false
+      ) {
+        return true;
+      }
+    }
+    return check(
+      false,
```

Comment:
Same here at this point, the values should only be for async hook keys.

Reply:
`Unexpected Node async_hooks additions to promise`

Done